src/System.Management.Automation/engine/InternalCommands.cs

```diff
@@ -2647,46 +2647,17 @@ public SwitchParameter Off
         private SwitchParameter _off;
 
         /// <summary>
-        /// To make it easier to specify a version, we add some conversions that wouldn't happen otherwise:
-        ///   * A simple integer, i.e. 2
-        ///   * A string without a dot, i.e. "2"
-        ///   * The string 'latest', which we interpret to be the current version of PowerShell.
+        /// Handle 'latest', which we interpret to be the current version of PowerShell.
         /// </summary>
-        private sealed class ArgumentToVersionTransformationAttribute : ArgumentTransformationAttribute
+        private sealed class CommandArgumentToVersionTransformationAttribute : ArgumentToVersionTransformationAttribute
```

Comment:
```suggestion
        private sealed class ArgumentToPSVersionTransformationAttribute : ArgumentToVersionTransformationAttribute
```

Reply:
We convert to Version, not to PSVersion.

Reply:
The use of this particular attribute is for `Set-StrictMode -Version`, and the argument must be valid PowerShell version.
```
> Set-StrictMode -Version 3.5
Set-StrictMode: Cannot validate argument on parameter 'Version'. The "3.5" argument does not contain a valid PowerShell version. Supply a valid version number and then try the command again.
```